agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -128,17 +130,53 @@ def apply
           warn "failed to sync #{service_pod.name} at #{service_pod.deploy_rev}: #{error}"
           warn error
           sync_state_to_master(@container, error)
+          return
         else
           @container_state_changed = false
-          sync_state_to_master(@container)
+        end
 
+        if service_pod.terminated?
           # Only terminate this actor after we have succesfully ensure_terminated the Docker container
           # Otherwise, stick around... the manager will notice we're still there and re-signal to destroy
-          self.terminate if service_pod.terminated?
+          self.terminate
```

Comment:
Subtle change here: no longer calls `sync_state_to_master` after the call to `destroy`. That's probably a good thing, the server won't care anymore, and it would always log an error when terminating service pods:

```
E, [2017-09-06T13:25:41.627773 #8] ERROR -- RpcServer: RuntimeError: Instance not found
D, [2017-09-06T13:25:41.627884 #8] DEBUG -- RpcServer: /app/app/services/rpc/node_service_pod_handler.rb:48:in `set_state'
/app/app/services/rpc_server.rb:63:in `handle_request'
/app/app/services/rpc_server.rb:47:in `process!'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/call/async.rb:7:in `dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
/usr/lib/ruby/gems/2.4.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
```